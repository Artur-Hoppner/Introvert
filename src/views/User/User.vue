<template>
  <v-card max-width="344" class="mx-auto">
    <v-list-item>
      <v-list-item-avatar color="grey"></v-list-item-avatar>
      <v-list-item-content>
        <v-list-item-title class="headline">User profile</v-list-item-title>
        <v-list-item-subtitle>by Kurt Wagner</v-list-item-subtitle>
      </v-list-item-content>
    </v-list-item>

    <v-img
      src="https://cdn.vuetifyjs.com/images/cards/mountain.jpg"
      height="194"
    ></v-img>

    <v-card-text>
      Email: {{ user.email }}Username: {{ user.username }}Name: {{ user.name }}
    </v-card-text>

    <v-card-actions>
      <v-btn text color="deep-purple accent-4">
        Read
      </v-btn>
      <v-btn text color="deep-purple accent-4">
        Bookmark
      </v-btn>
      <v-spacer></v-spacer>
      <v-btn icon>
        <v-icon>mdi-heart</v-icon>
      </v-btn>
      <v-btn icon>
        <v-icon color="red">mdi-share-variant</v-icon>
      </v-btn>
    </v-card-actions>
  </v-card>
</template>

<script>
import { mapGetters } from 'vuex';
export default {
  computed: mapGetters(['user'])
};
</script>

<style></style>
