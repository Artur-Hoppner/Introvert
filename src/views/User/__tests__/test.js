// Utilities
import { mount, createLocalVue } from '@vue/test-utils';
// Libraries
import Vue from 'vue';
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import Vuetify from 'vuetify';
// Components
import User from '@/views/User/User.vue';

const localVue = createLocalVue();
localVue.use(Vuex, VueRouter);

jest.mock('vuex-map-fields', () => ({
  mapFields: jest.fn()
}));

describe('/user', () => {
  const store = new Vuex.Store({
    state: { user: 'Artur' },
    getters: {
      user: state => state.user
    }
  });

  let vuetify;
  beforeEach(() => {
    Vue.use(Vuetify);
    vuetify = new Vuetify();
  });
  test('Test id User is rendered', () => {
    // Arrange;
    const wrapper = mount(User, {
      localVue,
      vuetify,
      store
    });
    // Assert
    expect(wrapper.element).toMatchSnapshot();
  });
});
