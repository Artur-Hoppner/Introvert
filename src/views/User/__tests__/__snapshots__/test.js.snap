// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/user Test id User is rendered 1`] = `
<div
  class="mx-auto v-card v-sheet theme--light"
  style="max-width: 500px;"
>
  <div
    class="v-image v-responsive white--text align-end theme--light"
    style="height: 240px;"
  >
    <transition-stub
      mode="in-out"
      name="fade-transition"
    >
      <div
        class="v-image__image v-image__image--preload v-image__image--cover"
        style="background-position: center center;"
      />
    </transition-stub>
    <div
      class="v-responsive__content"
    >
      <div
        class="v-card__title"
      >
        
      </div>
    </div>
  </div>
   
  <div
    class="v-card__subtitle pb-0"
  >
     
  </div>
   
  <div
    class="v-card__text text--primary"
  >
    <div>
      
    </div>
  </div>
</div>
`;
